src/cpp/security/authentication/PKIDH.cpp

```diff
@@ -1051,6 +1052,36 @@ ValidationResult_t PKIDH::validate_local_identity(
         password = &empty_password;
     }
 
+    std::string key_agreement_algorithm = ECDH_prime256v1;
+    std::string* key_agreement_property =
+            PropertyPolicyHelper::find_property(auth_properties, "preferred_key_agreement");
+    if (nullptr != key_agreement_property)
+    {
+        const std::pair<std::string, std::string> key_agreement_allowed_values[] = {
+            {DH_2048_256, DH_2048_256},
+            {ECDH_prime256v1, ECDH_prime256v1},
+            {"ECDH", ECDH_prime256v1},
+            {"RSA", DH_2048_256}
```

Comment:
It may be a good idea also to move these two "short alias" values to `PKIIdentityHandle.h` as `static const char* const`

Reply:
Not sure about this. `PKIIdentityHandle.h` is a private header, and the constants there represent the possible values that can be present in the `kagree_algo_` of the exchanged security messages.